tmt/base.py

```diff
@@ -1271,6 +1273,9 @@ def show(self):
             echo(tmt.utils.format(
                 'description', self.description, key_color='green'))
 
+        if self._import_plan_node:
+            self._import_plan_node.show()
```

Comment:
I'd probably rely on `self.imports_plan` and `self.imported_plan.show()`. Is there a reason why using the low-level info stored in the fmf node itself? I'd expect to work with higher-level classes like `Plan` at this place. After all, it's `Plan.show()` we're in at this point.

Also, I'd maybe move this step to the end of show, so it'd be presented in the "this is the plan, and this is the plan it imports" fashion - your code would sort of injecting the imported plan into the middle of the original plan, and especially the original plan's steps would be printed *after* the imported plan, suggesting they belong to that plan instead.

Reply:
Thanks for the tip. This piece of code should not exist. It was deleted during refactoring and was introduced again during rebase to main.

Reply:
One more thing. Currently with `[plan.imported_plan or plan for plan in plans]` the origin plan is not printed at all. Should be `Tree.plans` method parametrized to allow returning without without imported plans? If yes, then `Plan.show()` may also implement high-level details about FmfID about the imported plan or the whole plan if an additional `--deep` parameter is added.

Reply:
Maybe we could establish a bidirectional relationship: we have `plan.imported_plan`, but maybe we could also add `imported_plan`, or `origin_plan` or something, which would the *imported* plan use to point to the plan that imported it. Then in `Plan.show()`, we could have a short `if self.origin-plan: self.info('this was the original plan'); self.origin_plan.show()`.

Does it even make sense? :) In this picture, we'd need to add the link in the opposite direction, and then just display. we can even add an option to control this, `--show-plan-import-origins`, to not be too verbose by default :)

Reply:
I added a reference to the origin plan. Please have a look and share your thoughts. I would like to discuss the `--deep` or `--show-plan-import-origins` parameter also with @psss. 

The first question is whether to clone repo every single time and show it optionally. This can also be achieved by `--verbose` option.
The second question is how to conditionally clone repo for actions such as tmt run, tmt plan, tmt plan show, etc.

What do you think about this?

Reply:
What about the following approach: By default use the `fmf` cache for `tmt plan show` and friends and introduce a new option `--shallow` to disable fetching remote plans altogether for cases when it is really necessary to speed up things (like the huge kernel tests git repository ;-)

When `run` is attached to the plan we always do a full git clone because it means we're working on the plan execution which needs to have the latest repo fetched into the run workdir.

Reply:
Thanks, works nicely!